engine/core/source/org/pentaho/reporting/engine/classic/core/util/Worker.java

```diff
@@ -113,7 +115,10 @@ public void finish() {
       state = Worker.STATE_DEAD;
 
       try {
-        this.interrupt();
+        final AbstractReportProcessor processor = ReportProcessorThreadHolder.getProcessor();
```

Comment:
I would rather see the cancellation code moved upwards to the PreviewPane. Technically either location would work, but conceptually, if you see a worker, you'd expect it to be not bound to a specific runnable type. Just use "printReportProcessor.cancel()" within the PreviewPane and your cancellation request will be granted.

Within the PreviewPane, you have direct access to the report-processor, so you don't even need to use the ReportProcessorThreadHolder for the cancellation. This makes the intent of the code much clearer. 

Another advantage (unnoticeable to the ordinary user): You now no longer actually kill the worker (state = STATE_DEAD), so the thread will be reused for the next task.